pkg/controller/pkg/manager/digest.go

```diff
@@ -52,7 +52,12 @@ func NewPackageDigester(fetcher xpkg.Fetcher) *PackageDigester {
 func (d *PackageDigester) Digest(ctx context.Context, p v1alpha1.Package) (string, error) {
 	pullPolicy := p.GetPackagePullPolicy()
 	if pullPolicy != nil && *pullPolicy == corev1.PullNever {
-		return p.GetSource(), nil
+		return xpkg.FriendlyID(p.GetName(), p.GetSource()), nil
+	}
+	if pullPolicy != nil && *pullPolicy == corev1.PullIfNotPresent {
+		if p.GetCurrentIdentifier() == p.GetSource() {
```

Comment:
I'm confused about what this condition is checking for. It seems like it is checking if the digest method has successfully been computed yet. Once it has, `CurrentRevision` is set to the digest, and `CurrentIdentifier` is set equal to `Source`. I wonder if you could check if `CurrentRevision` is set instead, since that is more directly connected to the reconciler using the output of this function (and also the value you return in that condition)? As it currently stands I have to reason about why `CurrentIdentifier` being equal to `Source` is important, which is confusing because the function doesn't use `CurrentIdentifier`. I think the reason is because this is how you implement the "blank out `CurrentIdentifier` to recompute `CurrentRevision` hack, but maybe switching to `CurrentRevision` would enable the same thing and be more intuitive?

Reply:
See comment above about blanking out `CurrentIdentifier`, but we unfortunately cannot just use `CurrentRevision` here. Envision the following scenario:
- `Provider` is create with `package: crossplane/provider-gcp:v0.12.0` and `packagePullPolicy: IfNotPresent`
- We create a `ProviderRevision` with name `provider-gcp-506d40223706` (`CurrentRevision`)
- `CurrentIdentifier` is set to `crossplane/provider-gcp:v0.12.0`
- We later bump to `crossplane/provider-gcp:v0.13.0`. On the subsequent reconcile, we don't know what field changed, we just know there was a change. If `packagePullPolicy: IfNotPresent`, we don't know whether to check for the digest or not. 

An alternative option would be to look at the `ProviderRevision` that matches `CurrentRevision` and see if its `spec.source` matched the `spec.package` of the `Provider`. However, I'm not sure this is much of an improvement as you would still find yourself in the same situation if you lost your `status`.

Reply:
Got it, creating a new revision when the provider image url changes, sounds like a useful feature haha